File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int low = 0,high = nums.size()-1;
        int first=-1,last=-1;
        
        while(low<=high)             //binary search for first occurence of element
        { 
            int mid = low + (high-low)/2;
            if(nums[mid] == target)
            {
                first = mid;
                high = mid-1;
            }
            else if(nums[mid]>target)
            {
                high = mid-1;
            }
            else
            {
                low=mid+1;
            }
        }
        
        low=0,high=nums.size()-1;
        while(low<=high)             //binary search for last occurence of element
        { 
            int mid = low + (high-low)/2;
            if(nums[mid] == target)
            {
                last = mid;
                low = mid+1;
            }
            else if(nums[mid]>target)
            {
                high = mid-1;
            }
            else
            {
                low=mid+1;
            }
        }
        return {first,last};
    }
};

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int[] ans = new int[2];
        int low=0,high=nums.length-1;
        int ind = -1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(nums[mid] == target){
                ind = mid;
                high = mid-1;
            }
            else if(nums[mid] < target){
                low = mid+1;
            }
            else{
                high = mid-1;
            }
        }
        ans[0] = ind;
        low=0;
        high=nums.length-1;
        ind=-1;
        while(low<=high){
            int mid = low + (high-low)/2;
            if(nums[mid] == target){
                ind = mid;
                low = mid+1;
            }
            else if(nums[mid] < target){
                low = mid+1;
            }
            else{
                high = mid-1;
            }
        }
        ans[1]=ind;
        return ans;
    }
}